### DISCUSSION TITLE
Mailbox_Permissions_inventory.ps1  (RPS vs REST)

### DISCUSSION THREAD

**Discussion by swim709** [michevnew/PowerShell#13] [Q&A] [answered]
Hey there.

Thanks for putting all of these scripts together. I was wondering if you might be able to offer me any guidance for your script called "Mailbox_Permissions_inventory.ps1".

It seems that the only way I can get it to work is by using the -UseRPSSession option for  Connect-ExchangeOnline, before running the script.
However I saw that this option will no longer be available after June 2023.

I read that the REST cmdlets are supposed to be more or less interchangeable with the RPS cmdlets, so I feel that there is something simple I am missing here that would make it work going forward, without -UseRPSSession.

Admittedly I am not very experienced with this sort of scripting in PowerShell, but I can figure things out pretty well with just a little guidance  :)

**Comment by swim709** [comment, marked as answer]
Ah, I just noticed the V2 script  :)
That seems to be working great.
Thanks!

**Comment by michevnew** [comment]
I'm slowly updating my script samples to the latest and greatest, I suppose I should update the corresponding blog posts too. Too much to do... :)

**Reply by swim709** [reply]
I appreciate it!

Since you're here... I managed to use your  Remove_User_MBpermissions.ps1  by using the RPSSession, but it only removes Full Access permissions and the Send As + Send on Behalf permissions remain. Do you have any suggestions for handling those?

**Reply by michevnew** [reply]
That script only covers Full access permissions, I'll add the others when I get to the V2 version...
